src/games/wows/api.ts

```diff
@@ -49,41 +49,45 @@ export const getWowsApiData = async (token: string, setGameApi: (api: Api) => vo
     const response: WowsShipsApiResponse = await axios.get(`${process.env.VUE_APP_MS_WG_API}/wows/encyclopedia/ships/?page_no=${i}`, { headers })
     apiShips = apiShips.concat(Object.values(response.data.data))
   }
-  let ships: Ship[] = apiShips.map((ship: ApiShipResponse) => {
-    return {
-      id: `${ship.ship_id}`,
-      name: ship.name,
-      tier: ship.tier,
-      type: ship.type as WowsShipType,
-      game: Game.WOWS,
-      image: gameInfo.data.data.ship_type_images[ship.type][ship.is_special ? 'image_elite' : ship.is_premium ? 'image_premium' : 'image'],
-      canvasImage: {
-        image: canvasIcons[ship.type as WowsShipType].image,
-        dimensions: {
-          width: canvasIcons[ship.type as WowsShipType].width,
-          height: canvasIcons[ship.type as WowsShipType].height
-        }
-      },
-      default: false,
-      data: {
-        artillery: ship.default_profile.artillery?.distance,
-        torpedo: ship.default_profile.torpedoes?.distance,
-        atbas: ship.default_profile.atbas?.distance,
-        concealmentPlane: ship.default_profile.concealment?.detect_distance_by_plane,
-        concealmentShip: ship.default_profile.concealment?.detect_distance_by_ship,
-        hydro: undefined,
-        radar: undefined
+  let ships: Ship[] = apiShips.map((ship: ApiShipResponse): Ship => ({
+    id: `${ship.ship_id}`,
+    uuid: uuid(),
+    name: ship.name,
+    title: ship.name,
+    tier: ship.tier,
+    type: ship.type as WowsShipType,
+    game: Game.WOWS,
+    color: colors[ship.type],
+    image: gameInfo.data.data.ship_type_images[ship.type].image,
+    canvasImage: {
+      image: canvasIcons[ship.type as WowsShipType].image,
+      dimensions: {
+        width: canvasIcons[ship.type as WowsShipType].width,
+        height: canvasIcons[ship.type as WowsShipType].height
       }
-    } as Ship
-  })
+    },
+    default: false,
+    data: {
+      artillery: ship.default_profile.artillery?.distance,
+      torpedo: ship.default_profile.torpedoes?.distance,
+      atbas: ship.default_profile.atbas?.distance,
+      concealmentPlane: ship.default_profile.concealment?.detect_distance_by_plane,
+      concealmentShip: ship.default_profile.concealment?.detect_distance_by_ship,
+      hydro: undefined,
+      radar: undefined
+    }
+  }))
 
   const defaultShips: Ship[] = [
     {
       id: '1',
+      uuid: uuid(),
       name: 'CV',
+      title: 'Aircraft Carrier',
       tier: 0,
       type: WowsShipType.AIR_CARRIER,
       game: Game.WOWS,
+      color: colors[WowsShipType.AIR_CARRIER],
       image: 'https://glossary-wows-global.gcdn.co/icons/vehicle/types/AirCarrier/standard_84f55678325d4b492215390a7f0b43008f3947ab201502cd979dcf4c37633cf3.png',
```

Comment:
I think we should use something like static.gametactic.eu for this rather than trusting external sources.

Just in case.

Reply:
Yeah no problem, when the resource has been created we can swap them out